test/board_test.rb

```diff
@@ -0,0 +1,76 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/board'
+require './lib/ship'
+require 'pry'
+
+class BoardTest < Minitest::Test
+
+  def test_a_board_exists
+    board = Board.new
+
+    assert_instance_of Board, board
+  end
+
+  def test_valid_and_invalid_coordinates
+    board = Board.new
+
+    assert board.valid_coordinate?("A1")
+    refute board.valid_coordinate?("A5")
+  end
+
+  def test_valid_placement_of_ships
+    board = Board.new
+    cruiser = Ship.new("Cruiser", 3)
+    submarine = Ship.new("Submarine", 2)
+
+# placement has to be same length as ship
+    refute board.valid_placement?(cruiser, ["A1", "A2"])
+    refute board.valid_placement?(submarine, ["A2", "A3", "A4"])
+# coordinates must be consecutive
+    refute board.valid_placement?(cruiser, ["A1", "A2", "A4"])
+    refute board.valid_placement?(submarine, ["A1", "C1"])
+# coordinates can’t be diagonal
+    refute board.valid_placement?(cruiser, ["A1", "B2", "C3"])
+    refute board.valid_placement?(submarine, ["C2", "D3"])
+# if all the previous checks pass then the placement should be valid
+    #assert board.valid_placement?(submarine, ["A1", "A2"])
```

Comment:
WHOOPS! This part now fails. I can't believe I missed it...Working on solution. I wouldn't accept merge request yet unless you have solution.